lib/trie/trie.go

```diff
@@ -681,6 +708,10 @@ func retrieveFromBranch(branch *Node, key []byte) (value []byte) {
 // keys and a boolean indicating if all keys with the prefix were deleted
 // within the limit.
 func (t *Trie) ClearPrefixLimit(prefixLE []byte, limit uint32) (deleted uint32, allDeleted bool) {
+	defer func() {
+		t.handleTrackedDeltas(nil)
+	}()
+
```

Comment:
```suggestion
	defer t.handleTrackedDeltas(nil)
```

Reply:
Keeping it as is (so I don't forget like in a few hours on another branch) due to https://github.com/ChainSafe/gossamer/pull/2876#discussion_r989956510